lib/Sabre/PublicAlbumAuthBackend.php

```diff
@@ -21,46 +21,58 @@
  */
 namespace OCA\Photos\Sabre;
 
-use OC\Security\Bruteforce\Throttler;
-use OCA\Photos\Album\AlbumMapper;
-use OCP\IRequest;
-use Sabre\DAV\Auth\Backend\AbstractBasic;
+use Sabre\DAV\Auth\Backend\BackendInterface;
+use Sabre\HTTP\RequestInterface;
+use Sabre\HTTP\ResponseInterface;
 
-class PublicAlbumAuthBackend extends AbstractBasic {
-	private const BRUTEFORCE_ACTION = 'publicphotos_webdav_auth';
-	private IRequest $request;
-	private AlbumMapper $albumMapper;
-	private Throttler $throttler;
-
-	public function __construct(
-		IRequest $request,
-		AlbumMapper $albumMapper,
-		Throttler $throttler
-	) {
-		$this->request = $request;
-		$this->albumMapper = $albumMapper;
-		$this->throttler = $throttler;
+class PublicAlbumAuthBackend implements BackendInterface {
+	public function __construct() {
 	}
 
 	/**
-	 * Validates the token.
+	 * When this method is called, the backend must check if authentication was
+	 * successful.
+	 *
+	 * The returned value must be one of the following
+	 *
+	 * [true, "principals/username"]
+	 * [false, "reason for failure"]
+	 *
+	 * If authentication was successful, it's expected that the authentication
+	 * backend returns a so-called principal url.
+	 *
+	 * Examples of a principal url:
+	 *
+	 * principals/admin
+	 * principals/user1
+	 * principals/users/joe
+	 * principals/uid/123457
+	 *
+	 * If you don't use WebDAV ACL (RFC3744) we recommend that you simply
+	 * return a string such as:
+	 *
+	 * principals/users/[username]
 	 *
-	 * @param string $username
-	 * @return bool
-	 * @throws \Sabre\DAV\Exception\NotAuthenticated
+	 * @return array
 	 */
-	protected function validateUserPass($username, $password) {
-		$this->throttler->sleepDelayOrThrowOnMax($this->request->getRemoteAddress(), self::BRUTEFORCE_ACTION);
-
-		$albums = $this->albumMapper->getSharedAlbumsForCollaboratorWithFiles($username, AlbumMapper::TYPE_LINK);
-
-		if (count($albums) !== 1) {
-			$this->throttler->registerAttempt(self::BRUTEFORCE_ACTION, $this->request->getRemoteAddress());
-			return false;
-		}
-
-		\OC_User::setIncognitoMode(true);
+	public function check(RequestInterface $request, ResponseInterface $response) {
+		\OC_User::setIncognitoMode(true); // ???
```

Comment:
From the declaration: `set incognito mode, e.g. if a user wants to open a public link`

```suggestion
		\OC_User::setIncognitoMode(true);
```

Reply:
I am still not 100% sure what the `setIncognitoMode` call actually does. I suspect that it prevents using information about the logged-in user in further processing. However, I do not know how this applies in this case.